v3/src/components/slider/editable-slider-value.tsx

```diff
@@ -3,25 +3,26 @@ import {observer} from "mobx-react-lite"
 import React, {useState, useEffect} from "react"
 import {ISliderModel} from "./slider-model"
 import {MultiScale} from "../axis/models/multi-scale"
-import { AxisBounds } from "../axis/axis-types"
 
 import './slider.scss'
 
 interface IProps {
   sliderModel: ISliderModel
-  domain: AxisBounds | undefined
   multiScale: MultiScale
 }
 
-export const EditableSliderValue = observer(function EditableSliderValue({ sliderModel, domain, multiScale}: IProps) {
+export const EditableSliderValue = observer(function EditableSliderValue({ sliderModel, multiScale}: IProps) {
   const [candidate, setCandidate] = useState("")
+  const axisModel = sliderModel.axis
+  const axisMin = axisModel?.min
+  const axisMax = axisModel?.max
 
-  // when `domain` is not included in the dependency, slider value shows NaN
+  // when `multiScale.cellLength` is not included in the dependency, slider value shows NaN
   useEffect(() => {
     if (sliderModel) {
       setCandidate(multiScale.formatValueForScale(sliderModel.value))
     }
-  }, [domain, multiScale, sliderModel, sliderModel.axis, sliderModel.value])
+  }, [axisMin, axisMax, multiScale.cellLength, multiScale, sliderModel, sliderModel.value])
```

Comment:
This really wants to be a MobX reaction so that model dependencies are tracked automatically rather than having to be hand-crafted as `useEffect()` dependencies. I suspect something like this would work:
```typescript
  useEffect(() => {
    return autorun(() => {
      // trigger update on domain change
      const domain = sliderModel.domain
      if (sliderModel) {
        setCandidate(multiScale.formatValueForScale(sliderModel.value))
      }
    })
  }, [...])
```